sequencing-server/src/routes/command-expansion.ts

```diff
@@ -300,13 +303,117 @@ commandExpansionRouter.post('/expand-all-activity-instances', async (req, res, n
       `POST /command-expansion/expand-all-activity-instances: No expansion run was inserted in the database`,
     );
   }
-  const id = rows[0].id;
+  const expansionRunId = rows[0].id;
   logger.info(
-    `POST /command-expansion/expand-all-activity-instances: Inserted expansion run to the database: id=${id}`,
+    `POST /command-expansion/expand-all-activity-instances: Inserted expansion run to the database: id=${expansionRunId}`,
+  );
+
+  // Get all the sequence IDs that are assigned to simulated activities.
+  const seqToSimulatedActivity = await db.query(
+    `
+      select seq_id
+      from sequence_to_simulated_activity
+      where sequence_to_simulated_activity.simulated_activity_id in (${pgFormat(
+        '%L',
+        expandedActivityInstances.map(eai => eai.id),
+      )})
+      and simulation_dataset_id = $1
+    `,
+    [simulationDatasetId],
   );
 
+  if (seqToSimulatedActivity.rows.length > 0) {
+    const seqRows = await db.query(
+      `
+        select metadata, seq_id, simulation_dataset_id
+        from sequence
+        where sequence.seq_id in (${pgFormat(
+          '%L',
+          seqToSimulatedActivity.rows.map(row => row.seq_id),
+        )})
+        and sequence.simulation_dataset_id = $1;
+      `,
+      [simulationDatasetId],
+    );
+
+    // If the user has created a sequence, we can try to save the expanded sequences when an expansion runs.
+    if (seqRows.rows.length > 0) {
+      const seqId = seqRows.rows[0].seq_id;
```

Comment:
So instead of just saving a single sequence, we need to account for multiple sequences. For example try adding two sequences and expanding. We should see both in the DB.

Reply:
🤦 @camargo catching my oversite. Should be a simple enough map over seqRows.rows.

Reply:
This is implemented now, the only problem I had was getting `seq_id` setup as a foreign key, so right now the tables aren't officially linked. I tried just adding it like:
```
  foreign key (seq_id)
    references sequence (seq_id)
    on delete cascade
```
but that didn't work, any ideas?